### DISCUSSION TITLE
How to combine the "lang"-Option with the "babel"-package

### DISCUSSION THREAD

**Discussion by timonscheuer** [quarto-dev/quarto-cli#3844] [Q&A]
I‘m working on a research paper with R markdown using Quarto. I need both the `lang: de` option for the German textual output and my references and the `\usepackage[ngerman]{babel}`-option for automatic hyphenation to meet the requirements of my supervisor. Both options work separated, but not together. Using both creates an „Option clash for package babel“.

I was told by @mcanouil to modify a partial of the template, but as I am a beginner in R-Markdown and in Quarto (just decided to switch from word), I don't really know where to start.

So I decided to post my code here, maybe someone can help me.

In this version, without using `lang: de`, the automatic hyphenation works perfectly, but the textual output as well as some parts of my refernces are in English:

```
---
title: "Teamprojekt-Exposé"
author: "Timon Scheuer"
date: "01/08/2023"
#lang: de 
format:
  pdf:
    toc: true
    toc-title: "Inhalt"
    number-sections: true
    colorlinks: true
fontsize: 12pt
geometry: margin=3.5cm
usepackage: [scrlayer-scrpage]
header-includes:
  - \usepackage{fancyhdr}
  - \pagestyle{plain}
  - \fancyfoot[C]{\thepage}
  - \pagenumbering{arabic}
  - \setcounter{page}{0}
  - \usepackage{lmodern} # Optional newtxtext
  - \usepackage{sectsty}
  - \addtokomafont{disposition}{\rmfamily}
  - \usepackage{graphicx}
  - \usepackage[ngerman]{babel}
bibliography: Teamprojekt_Zotero.bib
csl: /Users/timon/Desktop/R/R_Projekte/Masterarbeit Test/apa-no-initials.csl
linestretch: 1.5
link-citations: true
date-format: "DD.MM.YYYY"
editor_options: 
  chunk_output_type: console
---

```{=tex}
\thispagestyle{empty}
\newpage
#```

# Fragestellung und Relevanz

Im Teamprojekt wird der Frage nachgegangen werden, welcher der im sozialpsychologischen Modell zur Erklärung von Wahlverhalten [vgl. einführend @schoenSozialpsychologischeAnsatzZur2014] angeführten Faktoren bei Wählern der verschiedenen Parteien den Ausschlag für die Wahlentscheidung bei der Bundestagswahl 2021 gegeben hat. Von besonderem Interesse ist hierbei die Frage, wie Wählerinnen und Wählern mit *cross-pressures* und den daraus resultierenden kognitiven Dissonanzen umgegangen sind und wie dies erklärt werden kann.

Die Relevanz dieser Fragestellung ergibt sich aus der parteipolitischen Konstellation vor der Bundestagswahl 2021, die sich in vielerlei Hinsicht von anderen Bundestagswahlen unterscheidet. Für den Kontext der hier vorzustellenden Arbeit seien drei Punkte besonders hervorgehoben: Zum einen trat zum ersten Mal in der Geschichte der Bundesrepublik trat eine amtierende Bundeskanzlerin nicht zur Wiederwahl an, was für die bis dahin die Bundeskanzerin stellenden Unionsparteien eine besondere Herausforderung und für alle anderen Parteien angesichts des fehlenden Amtsbonus eine besondere Chance darstellte.

Zum anderen starteten mit CDU/CSU, SPD und Bündnis 90/Die Grünen zum ersten Mal drei Parteien einen Kanzlerkandidaten beziehungsweise eine Kanzlerkandidatin benannt haben, die unterschiedlichste politische und biografische Hintergründe und dadurch auch verschiedene politische Erfahrungen und Kompetenzen vorzuweisen haben.


| Col1 | Col2 | Col3 |
|------|------|------|
|      |      |      |
|      |      |      |
|      |      |      |

: just a test


```{=tex}
\newpage
%%\setcounter{page}{26}
#```

# Literatur
```

<img width="793" alt="Bildschirm­foto 2023-01-04 um 11 54 47" src="https://user-images.githubusercontent.com/99292110/210540632-53fd5989-0875-4be9-8607-a174c90fe34b.png">

<img width="315" alt="Bildschirm­foto 2023-01-04 um 12 16 55" src="https://user-images.githubusercontent.com/99292110/210544028-c7523b48-4e79-4c94-a2de-f86c78193e03.png">

In this version, it's the other way around: The textual output works (including the refernces), but the hyphenation doesn't:

```
---
title: "Teamprojekt-Exposé"
author: "Timon Scheuer"
date: "01/08/2023"
lang: de
format:
  pdf:
    toc: true
    toc-title: "Inhalt"
    number-sections: true
    colorlinks: true
fontsize: 12pt
geometry: margin=3.5cm
usepackage: [scrlayer-scrpage]
header-includes:
  - \usepackage{fancyhdr}
  - \pagestyle{plain}
  - \fancyfoot[C]{\thepage}
  - \pagenumbering{arabic}
  - \setcounter{page}{0}
  - \usepackage{lmodern} # Optional newtxtext
  - \usepackage{sectsty}
  - \addtokomafont{disposition}{\rmfamily}
  - \usepackage{graphicx}
  #- \usepackage[ngerman]{babel}
bibliography: Teamprojekt_Zotero.bib
csl: /Users/timon/Desktop/R/R_Projekte/Masterarbeit Test/apa-no-initials.csl
linestretch: 1.5
link-citations: true
date-format: "DD.MM.YYYY"
editor_options: 
  chunk_output_type: console
---

```{=tex}
\thispagestyle{empty}
\newpage
#```

# Fragestellung und Relevanz

Im Teamprojekt wird der Frage nachgegangen werden, welcher der im sozialpsychologischen Modell zur Erklärung von Wahlverhalten [vgl. einführend @schoenSozialpsychologischeAnsatzZur2014] angeführten Faktoren bei Wählern der verschiedenen Parteien den Ausschlag für die Wahlentscheidung bei der Bundestagswahl 2021 gegeben hat. Von besonderem Interesse ist hierbei die Frage, wie Wählerinnen und Wählern mit *cross-pressures* und den daraus resultierenden kognitiven Dissonanzen umgegangen sind und wie dies erklärt werden kann.

Die Relevanz dieser Fragestellung ergibt sich aus der parteipolitischen Konstellation vor der Bundestagswahl 2021, die sich in vielerlei Hinsicht von anderen Bundestagswahlen unterscheidet. Für den Kontext der hier vorzustellenden Arbeit seien drei Punkte besonders hervorgehoben: Zum einen trat zum ersten Mal in der Geschichte der Bundesrepublik trat eine amtierende Bundeskanzlerin nicht zur Wiederwahl an, was für die bis dahin die Bundeskanzerin stellenden Unionsparteien eine besondere Herausforderung und für alle anderen Parteien angesichts des fehlenden Amtsbonus eine besondere Chance darstellte.

Zum anderen starteten mit CDU/CSU, SPD und Bündnis 90/Die Grünen zum ersten Mal drei Parteien einen Kanzlerkandidaten beziehungsweise eine Kanzlerkandidatin benannt haben, die unterschiedlichste politische und biografische Hintergründe und dadurch auch verschiedene politische Erfahrungen und Kompetenzen vorzuweisen haben.


| Col1 | Col2 | Col3 |
|------|------|------|
|      |      |      |
|      |      |      |
|      |      |      |

: just a test


```{=tex}
\newpage
#```

# Literatur
```

<img width="793" alt="Bildschirm­foto 2023-01-04 um 11 55 55" src="https://user-images.githubusercontent.com/99292110/210541421-f7e1efdc-6428-4378-8200-e511467caa25.png">

<img width="315" alt="Bildschirm­foto 2023-01-04 um 12 16 13" src="https://user-images.githubusercontent.com/99292110/210543923-5bff24bb-ef43-4f0e-8db4-91430d7e293d.png">

Sadly, I really need both of them working together, but I simply don't know what to do. 

I'd be very happy if someone could help me.

Thank you!

Update: Just discovered that I don't even need the babel-option. Just need a way to keep hyphenation while using `lang: de`.

**Comment by dragonstyle** [comment]
I'm not sure how to determine when hyphenation is working properly for your document, but have you tried setting `babel-lang` like:

```
babel-lang: ngerman
```

This might get things working for you (complete YAMl below):

````
---
title: "Teamprojekt-Exposé"
author: "Timon Scheuer"
date: "01/08/2023"
lang: de
babel-lang: ngerman
format:
  pdf:
    toc: true
    toc-title: "Inhalt"
    number-sections: true
    colorlinks: true
fontsize: 12pt
geometry: margin=3.5cm
usepackage: [scrlayer-scrpage]
header-includes:
  - \usepackage{fancyhdr}
  - \pagestyle{plain}
  - \fancyfoot[C]{\thepage}
  - \pagenumbering{arabic}
  - \setcounter{page}{0}
  - \usepackage{lmodern} # Optional newtxtext
  - \usepackage{sectsty}
  - \addtokomafont{disposition}{\rmfamily}
  - \usepackage{graphicx}
bibliography: Teamprojekt_Zotero.bib
csl: /Users/timon/Desktop/R/R_Projekte/Masterarbeit Test/apa-no-initials.csl
linestretch: 1.5
link-citations: true
date-format: "DD.MM.YYYY"
editor_options: 
  chunk_output_type: console
---

```{=tex}
\thispagestyle{empty}
\newpage
#```

# Fragestellung und Relevanz

Im Teamprojekt wird der Frage nachgegangen werden, welcher der im sozialpsychologischen Modell zur Erklärung von Wahlverhalten [vgl. einführend @schoenSozialpsychologischeAnsatzZur2014] angeführten Faktoren bei Wählern der verschiedenen Parteien den Ausschlag für die Wahlentscheidung bei der Bundestagswahl 2021 gegeben hat. Von besonderem Interesse ist hierbei die Frage, wie Wählerinnen und Wählern mit *cross-pressures* und den daraus resultierenden kognitiven Dissonanzen umgegangen sind und wie dies erklärt werden kann.

Die Relevanz dieser Fragestellung ergibt sich aus der parteipolitischen Konstellation vor der Bundestagswahl 2021, die sich in vielerlei Hinsicht von anderen Bundestagswahlen unterscheidet. Für den Kontext der hier vorzustellenden Arbeit seien drei Punkte besonders hervorgehoben: Zum einen trat zum ersten Mal in der Geschichte der Bundesrepublik trat eine amtierende Bundeskanzlerin nicht zur Wiederwahl an, was für die bis dahin die Bundeskanzerin stellenden Unionsparteien eine besondere Herausforderung und für alle anderen Parteien angesichts des fehlenden Amtsbonus eine besondere Chance darstellte.

Zum anderen starteten mit CDU/CSU, SPD und Bündnis 90/Die Grünen zum ersten Mal drei Parteien einen Kanzlerkandidaten beziehungsweise eine Kanzlerkandidatin benannt haben, die unterschiedlichste politische und biografische Hintergründe und dadurch auch verschiedene politische Erfahrungen und Kompetenzen vorzuweisen haben.


| Col1 | Col2 | Col3 |
|------|------|------|
|      |      |      |
|      |      |      |
|      |      |      |

: just a test


```{=tex}
\newpage
#```

# Literatur
````

**Reply by timonscheuer** [reply]
Hey @dragonstyle,

thanks for your reply!

Just tried it. 

Setting `babel-lang: ngerman` combinded with `lang: de` (as suggested by you), that leads to the hyphenation I want, but the contextual output is still in English. Same just with `babel-lang: ngerman`. And if I only use `lang: de`, the contexual Output is in German, but there's no hyphenation. 

The thing is that most of the supervisors at German Universties insist on using hyphenation. As German is a language with very long words and as the expected lengh of a paper is given in a number of pages most of the time, they don't want a text to be streched, so they insist on hyphenation to ensure equality.  

So I depend on a possibility to combine German contextual output with hyphenation in order to use quarto and R markdown for my work. 

So, do you think that there's another way to make that possible?

Thanks again!

**Reply by cderv** [reply]
@timonscheuer can you provide a reproducible example ? One I can run on my side. 

I am wondering how Pandoc itself behave with this to check that our handling on `lang` does not conflict with Pandoc's babel default configuration. 

Also, I would look out in the web for option to configure hyphenation.  I wonder if adding in header something like 
````latex
\babelprovide[hyphenrules=ngerman-x-latest]{ngerman}
````
would work.

Thanks

**Comment by timonscheuer** [comment]
@cderv I'll try to. After you copied that to RStudio in a new .qmd-file, you just need to remove the "#" in lines 1, 30, 66 and 69.

I disabled the `csl:` and the `bibliography:` line in the YAML-header because they would of course require specific files which I can't upload here and which are not not necessary to run the code and to solve my problem. To check the language of the contextual output, just use the table as described below. 

(Sorry for the many packages, this is just what I gut at the beginning all my documents because these are the packages I usually use for my work ;))

The two options this is all about are in lines 36 & 37. At least that's what Ive tried so far.

I hope that's reproducable. Let me know if I can do anything to help you.  

Thank you!

```
#```{r Start, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
library(rmarkdown)
library(tinytex)
library(geometry)
library(stargazer)
library(haven)
library(dplyr)
library(ggpubr)
library(tidyr)
library(ggplot2)
library(survey)
library(surveysd)
library(ggsurvey)
library(readxl)
library(DataEditR)
library(plotrix)
library(foreign)
library(jtools)
library(questionr)
library(esquisse)
library(openxlsx)
library(REAT)
library(reshape)
library(data.table)
library(reshape2)
library(ggrepel)
library(scales)
library(GDAtools)
# setwd()
#```

---
title: "Teamprojekt-Exposé"
author: "Timon Scheuer"
date: "01/08/2023"
#lang: de # That's what it's all about. 
#babel-lang: ngerman
format:
  pdf:
    toc: true
    toc-title: "Inhalt"
    number-sections: true
    colorlinks: true
fontsize: 12pt
geometry: margin=3.5cm, a4paper
usepackage: [scrlayer-scrpage]
header-includes:
  - \usepackage{fancyhdr}
  - \pagestyle{plain}
  - \fancyfoot[C]{\thepage}
  - \pagenumbering{arabic}
  - \setcounter{page}{0}
  - \usepackage{lmodern} # Optional newtxtext
  - \usepackage{sectsty}
  - \addtokomafont{disposition}{\rmfamily}
  - \usepackage{graphicx}
#bibliography: /Users/timon/Desktop/R/mylibrary.bib
#csl: /Users/timon/Desktop/R/R_Projekte/Masterarbeit Test/apa-no-initials.csl
linestretch: 1.5
link-citations: true
date-format: "DD.MM.YYYY"
editor_options: 
  chunk_output_type: console
---

#```{=tex}
\thispagestyle{empty}
\newpage
#```
# Fragestellung und Relevanz

Im Teamprojekt wird der Frage nachgegangen werden, welcher der im sozialpsychologischen Modell zur Erklärung von Wahlverhalten angeführten Faktoren bei Wählern der verschiedenen Parteien den Ausschlag für die Wahlentscheidung bei der Bundestagswahl 2021 gegeben hat. Von besonderem Interesse ist hierbei die Frage, wie Wählerinnen und Wählern mit *cross-pressures* und den daraus resultierenden kognitiven Dissonanzen umgegangen sind und wie dies erklärt werden kann.

Die Relevanz dieser Fragestellung ergibt sich aus der parteipolitischen Konstellation vor der Bundestagswahl 2021, die sich in vielerlei Hinsicht von anderen Bundestagswahlen unterscheidet. Für den Kontext der hier vorzustellenden Arbeit seien drei Punkte besonders hervorgehoben: Zum einen trat zum ersten Mal in der Geschichte der Bundesrepublik trat eine amtierende Bundeskanzlerin nicht zur Wiederwahl an, was für die bis dahin die Bundeskanzerin stellenden Unionsparteien eine besondere Herausforderung und für alle anderen Parteien angesichts des fehlenden Amtsbonus eine besondere Chance darstellte.

Zum anderen starteten mit CDU/CSU, SPD und Bündnis 90/Die Grünen zum ersten Mal drei Parteien einen Kanzlerkandidaten beziehungsweise eine Kanzlerkandidatin benannt haben, die unterschiedlichste politische und biografische Hintergründe und dadurch auch verschiedene politische Erfahrungen und Kompetenzen vorzuweisen haben.

| Col1 | Col2 | Col3 |
|------|------|------|
| Just |      |      |
| a    |      |      |
| test |      |      |

: Just a test for the contextual output. If this table is labled with "Tabelle 1" (instead of "Table 1") and there's hyphenation in the document, everything works as it should.

\newpage

# Literatur

```

**Reply by cderv** [reply]
Thanks. I'll try it. 

> After you copied that to RStudio in a new .qmd-file, you just need to remove the "#" in lines 1, 30, 66 and 69.

FWIW look at https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us on how to format correctly

**Reply by cderv** [reply]
How do I check hyphenation by the way ?

**Reply by timonscheuer** [reply]
> FWIW look at https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us on how to format correctly

Thank you, didn't know that!

To check hyphenation, take a look at the end of the lines in the PDF-document. I'll attach two examples. One with hyohenation, but English contextual output ("Table 1") and one without hypenation, but German contextual output ("Tabelle 1").

I hope thats helpful.

Thank you! (I'm relatively new here, and I must say that this really is a great community so far!


<img width="904" alt="Hyphenation, but English output" src="https://user-images.githubusercontent.com/99292110/217026789-634b50b0-7613-4b79-af74-5eff9c98c940.png">
<img width="904" alt="No hyphenation, but German output" src="https://user-images.githubusercontent.com/99292110/217026804-a67b5d7b-426c-44a5-80c6-1f6c4a799a64.png">

**Reply by cderv** [reply]
So looking closer to this, here are some comments on which I observe

When you do set `lang: de`, the `.tex` file will correctly have the babel command 
````
\ifLuaTeX
\usepackage[bidi=basic]{babel}
\else
\usepackage[bidi=default]{babel}
\fi
\babelprovide[main,import]{ngerman}
````

When you set only `babel-lang: ngerman` , you won't get those lines. They are included only when `lang` is set. So in this case, setting the option has no effect. 

When setting both, it is exactly like in `lang: de` only because `babel-lang` is already the correct value. 


Now regarding hyphenation: 

* When not setting any lang option, you'll observe some hyphenation. This is I believe the default hyphenation pattern for a document without **babel**
* When setting `lang: de`, the document with be using **babel** with `ngerman`, and it seems it changes the behavior by avoiding hyphenation at all cost. It seems to reduce fonts and spacing so that words are not split. Check your example end of line words. 

I also tried with a list of same word, using also your defined font size `fontsize: 12pt`

<details>
<summary>With `lang: de` </summary>

````markdown
---
title: "Teamprojekt-Exposé"
author: "Timon Scheuer"
lang: de # That's what it's all about.
format: pdf
keep-tex: true
fontsize: 12pt
---

übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
````

</details>

![image](https://user-images.githubusercontent.com/6791940/217214264-7bf2c0ae-e704-45a9-8513-35f5b57e40a0.png)


<details>
<summary>When not setting lang de and keeping fontsize </summary>

````markdown
---
title: "Teamprojekt-Exposé"
author: "Timon Scheuer"
# lang: de # That's what it's all about.
format: pdf
keep-tex: true
fontsize: 12pt
---

übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
übertragen 
````

</details>

![image](https://user-images.githubusercontent.com/6791940/217214390-c910fc8a-3cab-444e-aa3a-08c0cae0b4a8.png)


So I don't exactly how all this works, but I don't know if Quarto or even Pandoc has an issue. 

This should probably be studied at the LaTeX level to really understand how **babel** works

**Comment by Martin15135215** [comment]
I had the same problem with hyphenation, presented here in detail https://github.com/quarto-dev/quarto-cli/issues/5821#issue-1742457721

There is something wrong with tinytex, using the tex from my system package mangaer solved it. Mabye that can help you futher

**Comment by yann1cks** [comment]
I fixed the problem by adding `lang: de`, which will not add hyphenation for unknown reason. Additionally I added `\PassOptionsToPackage{ngerman}{babel}` to `include-in-header`, which corrects whatever issue there is with babel. That is why I do not think it is a general problem with tinytex, it seems to be an issue with the .tex file generated by Quarto.

**Reply by mcanouil** [reply]
Could you check the intermediate LaTeX file using `keep-tex: true` and check the babel related code?

**Reply by yann1cks** [reply]
Hi Mickaël, 

i build a little test, based on the first paragraphs of the German Wikipedia.

````
---
title: "German Test"
author: "Yannick"
keep-tex: true

lang: de
---

# Deutsche Wikipedia
Die deutschsprachige Wikipedia ist die Ausgabe der freien Online-Enzyklopädie Wikipedia in deutscher Sprache.

Sie wurde im März 2001 als erste weitere Sprachausgabe etwa zwei Monate nach der englischsprachigen Wikipedia gegründet.[1]

Was die Anzahl der Artikel betrifft, ist sie mit aktuell 2.831.339 Artikeln nach der englischsprachigen sowie der stark von computergenerierten Artikeln geprägten cebuanosprachigen Wikipedia-Ausgabe die drittgrößte Wikipedia-Ausgabe. Gemessen an der Anzahl der Artikel-Bearbeitungen, aber auch der Administratoren, Autoren und der besonders aktiven Autoren ist die deutschsprachige Wikipedia nach der englischsprachigen Wikipedia die zweitgrößte.[2] Die Anzahl der aktiven Autoren stieg stark bis 2007, sank daraufhin leicht[3] bis 2020 und steigt seither wieder leicht. 
````

If I remove the `lang` parameter hyphenation works fine, but everything else does not work, e.g. figure labels. In this case there is no reference of babel in the .tex file. 

So I have to add `lang: de`. In this case the hyphenation does not work. The word "cebuanosprachigen" shows the problem. Here are the lines of the .tex file that mention babel, plus some before/after (starts at line 130):

````
\makeatother
\makeatletter
\makeatother
\makeatletter
\makeatother
\ifLuaTeX
\usepackage[bidi=basic]{babel}
\else
\usepackage[bidi=default]{babel}
\fi
\babelprovide[main,import]{ngerman}
% get rid of language-specific shorthands (see #6817):
\let\LanguageShortHands\languageshorthands
\def\languageshorthands#1{}
````

If I add `\PassOptionsToPackage{ngerman}{babel}` this get added at line 80 of the .tex file.

**Reply by cderv** [reply]
Thanks for sharing. I think this complement my digging from https://github.com/quarto-dev/quarto-cli/discussions/3844#discussioncomment-4891879

## TL;DR

`hyphen-german` needs to be installed as it is not by default, and nothing triggers its installation automatically 

````
> tlmgr info hyphen-german
package:     hyphen-german
category:    TLCore
shortdesc:   German hyphenation patterns.
longdesc:    Hyphenation patterns for German in T1/EC and UTF-8 encodings, for traditional and reformed spelling, including Swiss German. The package includes the latest patterns from dehyph-exptl (known to TeX under names 'german', 'ngerman' and 'swissgerman'), however 8-bit engines still load old versions of patterns for 'german' and 'ngerman' for backward-compatibility reasons. Swiss German patterns are suitable for Swiss Standard German (Hochdeutsch) not the Alemannic dialects spoken in Switzerland (Schwyzerduetsch). There are no known patterns for written Schwyzerduetsch.
installed:   No
sizes:       run: 2133k
relocatable: Yes
collection:  collection-langgerman
````

So do `tlmgr install hyphen-german` and it will work. 

This is also required for using the header-include part (also see below that it does not really work without)

## Details

After re-reading the PDF doc from https://ctan.org/pkg/babel, I understand that 

* `\PassOptionsToPackage{ngerman}{babel}` will use the `ngerman.ldf` file and requires the `babel-german` CTAN package https://www.ctan.org/pkg/babel-german
* `\babelprovide[main,import]{ngerman}` will use the INI version `babel-de.ini`

It seems to me that the INI version does not define any hyphenation. If we do 
```
format:
  pdf:
    latex-clean: false
```
to get access to the .log file we can see when no adding the `\PassOptionsToPackage{ngerman}{babel}` : 
````
Package babel Info: Importing data for ngerman
(babel)             from babel-de.ini. Reported on input line 131.
Package babel Info: Hyphen rules for 'ngerman' set to \l@nil
(babel)             (\language7). Reported on input line 131.
````

`\l@nil` is empty element. So I believe no hyphenation is found. 

When using `ngerman` option with babel, it will use the `babel-german` package, and when doing that I can see in the log
````
Package babel Warning: No hyphenation patterns were preloaded for
(babel)                the language 'German (new orthography)' into the format.
(babel)                Please, configure your TeX system to add them and
(babel)                rebuild the format. Now I will use the patterns
(babel)                preloaded for \language=0 instead on input line 48.

Package babel Info: Hyphen rules for 'ngerman' set to \l@english
(babel)             (\language0). Reported on input line 49.
````

So again, it is not finding specific rule an use english one. Probably not what you expect. 

Looking more deeply in how hyphenation works with LaTeX, here are some findings: 

I understand from some search that hyphenation are maintained in other tools and `hyphen-german` is one of them. I believe this is the one missing. If you install it (`tlmgr install hyphen-german`) then you will see some hyphenation. No more message thrown so I believe hyphenation patterns are found. 

I just don't know to which hyphenation rules `hyphen-german` is about. It seems German Hyphenation have different existing rules depending on period.  This is from `babel-german` doc at https://www.ctan.org/pkg/babel-german about hyphenation. And from https://ctan.org/topic/hyphenation, we can find different package like the [dehyph-exptl](https://ctan.org/pkg/dehyph-exptl) or [dehyph](https://ctan.org/pkg/dehyph) or [hyph-utf8](https://ctan.org/pkg/hyph-utf8). 

However, printing information about `hyphen-german` looks like it is the good one to have
````
> tlmgr info hyphen-german
package:     hyphen-german
category:    TLCore
shortdesc:   German hyphenation patterns.
longdesc:    Hyphenation patterns for German in T1/EC and UTF-8 encodings, for traditional and reformed spelling, including Swiss German. The package includes the latest patterns from dehyph-exptl (known to TeX under names 'german', 'ngerman' and 'swissgerman'), however 8-bit engines still load old versions of patterns for 'german' and 'ngerman' for backward-compatibility reasons. Swiss German patterns are suitable for Swiss Standard German (Hochdeutsch) not the Alemannic dialects spoken in Switzerland (Schwyzerduetsch). There are no known patterns for written Schwyzerduetsch.
installed:   No
sizes:       run: 2133k
relocatable: Yes
collection:  collection-langgerman
````

Overall, this is not that obvious what to use in which context, and it is for sure a complexity of TeX. There is a page for hyphenation in TeX at https://hyphenation.org/


Hope it helps
